chia/simulator/full_node_simulator.py

```diff
@@ -144,3 +203,288 @@ async def reorg_from_index_to_new_index(self, request: ReorgProtocol):
+        an address.
+
+        Arguments:
+            count: The number of blocks to process.
+            farm_to: The puzzle hash to farm the block rewards to.
+
+        Returns:
+            The total number of reward mojos for the processed blocks.
+        """
+        if isinstance(timeout, _Default):
+            timeout = count * timeout_per_block
+            timeout += 1
+
+        with anyio.fail_after(delay=timeout):
+            rewards = 0
+
+            if count == 0:
+                return rewards
+
+            for _ in range(count):
+                if guarantee_transaction_blocks:
+                    await self.farm_new_transaction_block(FarmNewBlockProtocol(farm_to))
+                else:
+                    await self.farm_new_block(FarmNewBlockProtocol(farm_to))
+                # hopefully there is no race between the
+                height = self.full_node.blockchain.get_peak_height()
+
+                if height is None:
+                    raise RuntimeError("Peak height still None after processing at least one block")
+
+                rewards += calculate_pool_reward(height) + calculate_base_farmer_reward(height)
+
+            return rewards
+
+    async def farm_blocks_to_wallet(
+        self,
+        count: int,
+        wallet: Wallet,
+        timeout: Union[None, _Default, float] = default,
+    ) -> int:
+        """Farm the requested number of blocks to the passed wallet. This will
+        process additional blocks as needed to process the reward transactions
+        and also wait for the rewards to be present in the wallet.
+
+        Arguments:
+            count: The number of blocks to farm.
+            wallet: The wallet to farm the block rewards to.
+
+        Returns:
+            The total number of reward mojos farmed to the requested address.
+        """
+        if isinstance(timeout, _Default):
+            timeout = (count + 1) * timeout_per_block
+            timeout += 5
+
+        with anyio.fail_after(delay=timeout):
+            if count == 0:
+                return 0
+
+            target_puzzlehash = await wallet.get_new_puzzlehash()
+            rewards = 0
+
+            block_reward_coins = set()
+            expected_reward_coin_count = 2 * count
+
+            # TODO: why two final transaction blocks and not just one?
+            for to_wallet in [*([True] * count), False, False]:
+                if to_wallet:
+                    rewards += await self.farm_blocks_to_puzzlehash(
+                        count=1,
+                        farm_to=target_puzzlehash,
+                        guarantee_transaction_blocks=False,
+                        timeout=None,
+                    )
+                else:
+                    await self.farm_blocks_to_puzzlehash(count=1, guarantee_transaction_blocks=True, timeout=None)
+
+                peak_height = self.full_node.blockchain.get_peak_height()
+                if peak_height is None:
```

Comment:
not possible

Reply:
Adding the assertion here as you want.  I still think that the assertion is an indication of an opportunity for improvement elsewhere.  It will take a bit of work to unravel this one though.